fendermint/vm/topdown/proof-service/src/f3_client.rs

```diff
@@ -0,0 +1,259 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+//! F3 client wrapper for certificate fetching and validation
+//!
+//! Wraps the F3 light client to provide:
+//! - Certificate fetching from F3 RPC
+//! - Full cryptographic validation (BLS signatures, quorum, chain continuity)
+//! - Sequential state management for validated certificates
+
+use crate::observe::{F3CertificateFetched, F3CertificateValidated, OperationStatus};
+use crate::types::ValidatedCertificate;
+use anyhow::{Context, Result};
+use filecoin_f3_lightclient::{LightClient, LightClientState};
+use ipc_observability::emit;
+use std::sync::Arc;
+use std::time::Instant;
+use tokio::sync::Mutex;
+use tracing::{debug, error, info};
+
+/// F3 client for fetching and validating certificates
+///
+/// Uses the F3 light client for:
+/// - Direct F3 RPC access
+/// - Full cryptographic validation (BLS signatures, quorum, continuity)
+/// - Stateful sequential validation
+pub struct F3Client {
+    /// Light client for F3 RPC and cryptographic validation
+    /// Using Mutex to allow async methods
```

Comment:
Why Mutex is needed? the access should be serialized by the owner.

Reply:
Mutex is used for interior mutability with &self methods. but since access is serialized by the service, we could try to refactor to use &mut self methods and remove both Arc and Mutex